pgjdbc/src/main/java/com/yugabyte/ysql/ClusterAwareLoadBalancer.java

```diff
@@ -266,28 +307,48 @@ public synchronized boolean refresh(Connection conn) throws SQLException {
     }
     // else clear server list
     long currTime = System.currentTimeMillis();
-    servers = getCurrentServers(conn);
-    if (servers == null) {
-      return false;
-    }
     lastServerListFetchTime = currTime;
     long now = System.currentTimeMillis() / 1000;
     long failedHostTTL = Long.getLong("failed-host-ttl-seconds", DEFAULT_FAILED_HOST_TTL_SECONDS);
     Set<String> possiblyReachableHosts = new HashSet();
     for (Map.Entry<String, Long> e : unreachableHosts.entrySet()) {
       if ((now - e.getValue()) > failedHostTTL) {
+        LOGGER.fine("Putting host  " + e.getKey() + " into possiblyReachableHosts");
         possiblyReachableHosts.add(e.getKey());
       } else {
         LOGGER.fine("Not removing this host from unreachableHosts: " + e.getKey());
       }
     }
+
+    boolean emptyHostToNumConnMap=false;
     for (String h : possiblyReachableHosts) {
       unreachableHosts.remove(h);
+      emptyHostToNumConnMap=true;
+    }
+
+    if(emptyHostToNumConnMap==true && !hostToNumConnMap.isEmpty()){
+      LOGGER.fine("clearing host to num conn map " + hostToNumConnMap.keySet());
+      for(String h : hostToNumConnMap.keySet()) {
+        hostToNumConnCount.put(h,hostToNumConnMap.get(h));
+      }
+      LOGGER.fine("hosts in host to num conn count " + hostToNumConnMap.toString());
```

Comment:
Did you mean to print hostToNumConnCount?
`"hosts in hostToNumConnCount: " + hostToNumConnCount`